### PR TITLE
feat(sync): Limit the maximum height of once sync

### PR DESCRIPTION
**Pull request by yejiayu**
**What type of PR is this?**
feat

**What this PR does / why we need it**:
Each successful download of 50 blocks will complete synchronization to avoid downloading too many blocks from a single node, which will affect the success rate of network transmission.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/consensus/src/synchronization.rs [hunk 1]
```diff
@@ -165,9 +148,17 @@ impl<Adapter: SynchronizationAdapter> OverlordSynchronization<Adapter> {
         current_height: u64,
         remote_height: u64,
     ) -> ProtocolResult<()> {
+        let remote_height = if current_height + ONCE_SYNC_BLOCK_LIMIT > remote_height {
+            remote_height
+        } else {
+            remote_height + ONCE_SYNC_BLOCK_LIMIT
```
Comment (LycrusHamster) [inline core/consensus/src/synchronization.rs:154]: current_height + ONCE_SYNC_BLOCK_LIMIT

**Comment by yejiayu** [comment]
/hold 

wait tests

**Comment by yejiayu** [comment]
/hold cancel

test pass

**Pull request merged**